cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/generator/GenerationProcessorImpl.java

```diff
@@ -603,6 +603,13 @@ private void generateTemplateAndWriteFile(File output, Template template, TextTe
     try (Writer out = new StringWriter()) {
       templateEngine.process(template, model, out, outputCharset);
       FileUtils.writeStringToFile(output, out.toString(), outputCharset);
+      if (output.getAbsolutePath().endsWith(".java")) {
```

Comment:
If possible, try to change the File output from generateTemplateAndWriteFile to Path.